src/main/java/com/cloudbees/jenkins/plugins/sshagent/RemoteAgentFactory.java

```diff
@@ -51,25 +50,15 @@
      */
     public abstract boolean isSupported(Launcher launcher, TaskListener listener);
 
-    @Deprecated
-    public RemoteAgent start(Launcher launcher, TaskListener listener) throws Throwable {
-        return start(launcher, listener, null);
-    }
-
     /**
      * Start a ssh-agent on the specified launcher.
      *
-     * @param launcher the launcher on which to start a ssh-agent.
+     * @param launcherProvider provides launchers on which to start a ssh-agent.
      * @param listener a listener for any diagnostics.
      * @param temp a temporary directory to use; null if unspecified
      * @return the agent.
      * @throws Throwable if the agent cannot be started.
      */
-    public /*abstract*/ RemoteAgent start(Launcher launcher, TaskListener listener, @CheckForNull FilePath temp) throws Throwable {
```

Comment:
I know I know, this breaks API comparability. However that was already done in #33, if that is any excuse. Same goes for the deprecated method above. To be honest, there would be quite a lot of hoop jumping in order to retain backwards compatability. But if it is a requirement, let me know and I'll fix it.

Reply:
I would not bother.